src/Metrics.h

```diff
@@ -235,12 +237,27 @@ class TopN final : public Metric
     datasketches::frequent_items_sketch<T> _fi;
     size_t _top_count = 10;
     std::string _item_key;
+    double _percentile;
+
+    uint64_t _get_threshold() const
+    {
+        auto items = _fi.get_frequent_items(datasketches::frequent_items_error_type::NO_FALSE_NEGATIVES);
+        datasketches::kll_sketch<uint64_t> quantile;
+        for (uint64_t i = 0; i < std::min(_top_count, items.size()); i++) {
+            quantile.update(items[i].get_estimate());
+        }
+        if (quantile.is_empty()) {
+            return 0;
+        }
+        return quantile.get_quantile(_percentile);
+    }
 
 public:
-    TopN(std::string schema_key, std::string item_key, std::initializer_list<std::string> names, std::string desc)
+    TopN(std::string schema_key, std::string item_key, std::initializer_list<std::string> names, std::string desc, uint64_t percentile = KLL_DEFAULT)
         : Metric(schema_key, names, std::move(desc))
         , _fi(MAX_FI_MAP_SIZE, START_FI_MAP_SIZE)
         , _item_key(item_key)
+        , _percentile(static_cast<double>(percentile) / 100)
```

Comment:
we probably want to add a check here to make sure the percentile is between 0 and 1.0 and throw if not

Reply:
done